src/PostgREST/ApiRequest.hs

```diff
@@ -189,12 +202,26 @@ userApiRequest confSchemas rootSpec req reqBody
         json <- csvToJson <$> CSV.decodeByName reqBody
         note "All lines must have same number of fields" $ payloadAttributes (JSON.encode json) json
       (CTOther "application/x-www-form-urlencoded", _) ->
-        let json = M.fromList . map (toS *** JSON.String . toS) . parseSimpleQuery $ toS reqBody
+        let json = mapFromListWithMultiple . map (toS *** JSON.String . toS) . parseSimpleQuery $ toS reqBody
             keys = S.fromList $ M.keys json in
         Right $ ProcessedJSON (JSON.encode json) keys
       (ct, _) ->
         Left $ toS $ "Content-Type not acceptable: " <> toMime ct
-  rpcPrmsToJson = ProcessedJSON (JSON.encode $ M.fromList $ second JSON.toJSON <$> rpcQParams) (S.fromList $ fst <$> rpcQParams)
+  rpcPrmsToJson = ProcessedJSON (JSON.encode $ mapFromListWithMultiple $ second JSON.toJSON <$> rpcQParams) (S.fromList $ fst <$> rpcQParams)
+  mapFromListWithMultiple ls = M.mapWithKey listToValue $ M.fromListWith reverseConcat . map valueToList $ ls
```

Comment:
This function is a tad complicated. We should try to simplify it.

I was remembering this code and I found out that I left this unused type:
https://github.com/PostgREST/postgrest/blob/ccad9eb9bc87b09de668eee8d9fb56aefc8862f9/src/PostgREST/Types.hs#L402

Maybe you could try and make that a `data RpcQParam` and add an [instance ToJSON](https://hackage.haskell.org/package/aeson-1.5.4.1/docs/Data-Aeson.html#t:ToJSON) with some of the `mapFromListWithMultiple` logic.

Also, I've found [jsonAccum](https://hackage.haskell.org/package/aeson-1.5.4.1/docs/Data-Aeson-Parser-Internal.html#v:jsonAccum) through https://github.com/bos/aeson/pull/714. Looks like it could help but unfortunately I haven't been able to find an example snippet of how to use it.

Reply:
`jsonAccum` didn't help, because it parses a string as json - we already have lists of params at that point.

However the other suggestion turned out to be great - simplifies the logic a lot!